### ISSUE TITLE
Missing features 

### ISSUE THREAD

**Issue by traversaro** [traversaro/vcpkg-binary-installer-generator#4]
To use the `vcpkg-binary-installer-generator` with the `QtIFW` CPack backend to produce YARP and ICUB binaries installer, we need to make sure that we implement two features that we have in the existing NSIS installers:
* Optionally modify the environmental variables to make sure that the installer libraries can be found. In my tests, it will be sufficient to add `<prefix>` and `<prefix>/debug` to the `CMAKE_PREFIX_PATH` enviromental variables, and `<prefix>/tools` to `CMAKE_PROGRAM_PATH`. 
* Support for quietly install the contents of the installer from the command line. This is particularly useful for CI system such as AppVeyor. In particular, if we plan to support the **online** installer, it would be particular useful to have the support for specifying the specific libraries that you want to install from the command line. 

cc @drdanz

**Comment by drdanz** [comment]
I think that `<prefix>/bin` should go to `PATH` as well. See also https://github.com/Microsoft/vcpkg/issues/1176

**Comment by drdanz** [comment]
About quiet install support, see robotology/yarp@6ddffaf54811c6528b2e67042d3aac54b0ca15c1 and
https://raw.githubusercontent.com/robotology-dependencies/robotology-dep-scripts/master/qt5/qt-installer-noninteractive.qs for Qt5 installer

**Comment by traversaro** [comment]
Related: https://bugreports.qt.io/browse/QTIFW-166 . 
I think it is not possible to to pass the list of components to install from the command line, but each project can use its own script to install the desired dependencies.

**Comment by traversaro** [comment]
Related: a vcpkg PR adding a IFW export type has been proposed in https://github.com/Microsoft/vcpkg/pull/1734 .

**Comment by traversaro** [comment]
Related project: https://dad.podsvirov.pro/ .

**Comment by podsvirov** [comment]
Hello @traversaro, please review latest changes in Microsoft/vcpkg#1734.

**Comment by traversaro** [comment]
Both features kinda of implemented in https://github.com/robotology-playground/vcpkg-binary-installer-generator/commit/e3fbfb1cf03eac4067d53dd2d3d7521e904fe9cd . 

* Env variables : handled by the scripts https://github.com/robotology-playground/vcpkg-binary-installer-generator/blob/improv/addPathsToEnvVariables.ps1 and https://github.com/robotology-playground/vcpkg-binary-installer-generator/blob/improv/removePathsFromEnvVariables.ps1 , that are automatically called by the QtIFW installer 
* Silent installation: implemented for QtIFW installer by passing launching it with the `-script` option and passing https://github.com/robotology-playground/vcpkg-binary-installer-generator/blob/improv/silent-install.qs : however (even if I did not tested it) this will not work on headless systems, because QtIFW always creates the windows. As an alternative solution, I also generate `ZIP` tarballs, that contain the powershell script for manipulating the paths. The user can manually extract the tarball and call the scripts to make the libraries available in his system. (Note: for an unknown reason, calling the powershell script to modify the env variables is not currentl working in AppVeyor).

**Comment by diegoferigo** [comment]
@traversaro On a headless system you can create a script that presses buttons and fills the fields. You can find an example for QtCreator [here](https://github.com/diegoferigo/development-iit/blob/master/Tools/QtCreatorSetup.js).

You can then use it as follows:

https://github.com/diegoferigo/development-iit/blob/43281b81b6b96ad67485c8a75668b4ada810baad/Tools/Dockerfile#L90-L94

**Comment by traversaro** [comment]
@diegoferigo Exactly, this is what I described in the previous comment and committed in the repo at https://github.com/robotology-playground/vcpkg-binary-installer-generator/blob/improv/silent-install.qs .
 What I do not know if that will work on (Windows) headless systems such as AppVeyor, as the windows is still created, but just hidden (see http://blog.qt.io/blog/2015/04/07/qt-installer-framework-2-0-released/#comment-1195532 ) .
